### DISCUSSION TITLE
Not able to apply set method on list of dataclass value

### DISCUSSION THREAD

**Discussion by saife245** [pydantic/pydantic#3792] [Question]
I want to apply the set method to get the unique value of a list of data class values.

list look like :[Value(roll=2, name='saif', id=23), Value(roll=2, name='saif', id=23), Value(roll=2, name='saif', id=23)]

The data class looks like this: 
`@dataclass
class Value:
    roll: int
    name: str
    id: int`

can anybody help m with this? thanks in advance.

Note: for loop can't be applied, as it is quite expensive. It compares the 50k value in the list on each iteration.

**Comment by sarangsbabu367** [comment]
@saife245 This is not related to pydantic. Anyway you can define a `__hash__` method for your dataclass, which will be used to compare instances of the class while forming set. 

Check this: https://stackoverflow.com/questions/52390576/how-can-i-make-a-python-dataclass-hashable-without-making-them-immutable

This is a possibility,

```python
from dataclasses import dataclass

@dataclass
class Value:
    roll: int
    name: str
    id: int

    def __hash__(self):
        return hash((self.roll, self.name, self.id))


x = [Value(roll=2, name='saif', id=23), Value(roll=2, name='saif', id=23), Value(roll=2, name='saif', id=23)]
y = [Value(roll=3, name='saif', id=23), Value(roll=2, name='saif', id=23), Value(roll=2, name='saif', id=23)]

print(set(x))
print(set(y))
```

```
{Value(roll=2, name='saif', id=23)}
{Value(roll=2, name='saif', id=23), Value(roll=3, name='saif', id=23)}
```
